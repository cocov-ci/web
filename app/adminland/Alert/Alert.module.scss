@import "styles/variables.module";

.alert {
  padding: 15px 80px 15px 15px;
  display: flex;
  border: 1px solid #e6d341;
  border-radius: 8px;
  align-items: center;
  min-height: 54px;
  background-color: #f4eebd;
  position: relative;
  overflow: hidden;

  p {
    margin-bottom: 0;
    z-index: 1;
  }

  .link {
    text-decoration: underline;
  }

  .icon {
    color: #e6d34192;
    right: -15px;
    position: absolute;
    bottom: -12px;
    z-index: 0;
  }
}
