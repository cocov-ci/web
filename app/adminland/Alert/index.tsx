import { AlertTriangle } from 'lucide-react'
import React from 'react'

import Text from 'app/common/Text'

import styles from './Alert.module.scss'

interface AlertProps {
  children: React.ReactNode
}

const Alert = ({ children }: AlertProps) => {
  return (
    <div className={styles.alert}>
      <AlertTriangle className={styles.icon} size="64" />
      <Text variant="description">{children}</Text>
    </div>
  )
}

export default Alert
