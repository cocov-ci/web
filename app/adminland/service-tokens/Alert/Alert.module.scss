@import "styles/variables.module";

.alert {
  padding: 15px;
  display: flex;
  margin-bottom: 10px;
  border: 1px solid #E6D341;
  border-radius: 8px;
  align-items: center;
  height: 54px;
  background-color: #F4EEBD;
  position: relative;
  overflow: hidden;

  p {
    margin-bottom: 0;
  }

  .link {
    text-decoration: underline;
  }

  .icon {
    color: opacify(#E6D341, 0.3);
    right: 0;
    position: absolute;
    top: 2px;
  }
}
