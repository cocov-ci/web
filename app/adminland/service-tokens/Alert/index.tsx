import { AlertTriangle, Info } from "lucide-react";
import Link from 'next/link'

import Text from 'app/common/Text'

import styles from './Alert.module.scss'

const Alert = () => {
  return (
    <div className={styles.alert}>
      <Text variant="description">
        <strong>Caution:</strong> Unlike User Tokens, Service Tokens have irrestrict access to all repository and data stored on this instance. Handle those tokens as passwords.
      </Text>
      <AlertTriangle className={styles.icon} size="64" />
    </div>
  )
}

export default Alert
