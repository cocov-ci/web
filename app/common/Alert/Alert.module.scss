@import "styles/variables";

.alert {
  display: flex;
  justify-content: center;
  position: relative;

  .box {
    border-radius: 8px;
    border: 1px solid $colorGray;
    padding: 29px 40px 70px 28px;
    width: 385px;
    display: flex;
    position: relative;
    z-index: 1;

    .content {
      flex: 1;
    }

    h2 {
      font-size: $fontTitle;
      font-variation-settings: "wght" 600;
      color: $colorGrayDark;
    }

    h3 {
      font-size: $fontDescription;
      font-variation-settings: "wght" 400;
      color: $colorGrayDark;
    }

    .icon {
      stroke-width: 1px;
      margin-top: 2px;
      margin-right: 22px;
      color: $colorGrayDark;
    }
  }
}
