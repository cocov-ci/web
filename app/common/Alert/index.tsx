import classNames from 'classnames'
import { LucideIcon } from 'lucide-react'
import React from 'react'

import styles from './Alert.module.scss'

type BoxProps = {
  children?: React.ReactNode
  className?: string
  title?: string
  description?: string
  icon?: LucideIcon
}

const Alert = ({ className, icon, title, description }: BoxProps) => {
  const Icon = icon as LucideIcon

  return (
    <div className={classNames(styles.alert, className)}>
      <div className={styles.box}>
        {icon && <Icon className={styles.icon} size={40} />}
        <div className={styles.content}>
          {title && <h2>{title}</h2>}
          {description && <h3>{description}</h3>}
        </div>
      </div>
    </div>
  )
}

export default Alert
