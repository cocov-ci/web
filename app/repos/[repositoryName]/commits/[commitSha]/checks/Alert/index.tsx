import { Info } from 'lucide-react'
import Link from 'next/link'

import Text from 'app/common/Text'

import styles from './Alert.module.scss'

const Alert = () => {
  return (
    <div className={styles.alert}>
      <Text gutterBottom variant="description">
        <strong>Notice:</strong> Those checks have already finished. See its
        results on the{' '}
        <Link className={styles.link} href="#">
          commit results page
        </Link>
        .
      </Text>
      <Info className={styles.icon} size="64" />
    </div>
  )
}

export default Alert
