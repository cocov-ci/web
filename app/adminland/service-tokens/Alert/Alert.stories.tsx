import type { Meta, StoryObj } from '@storybook/react'

import Alert from './index'

const meta: Meta<typeof Alert> = {
  title: 'Adminland/ServiceTokens/Alert',
  component: Alert,
  tags: ['autodocs'],
}

export default meta

type Story = StoryObj<typeof Alert>

export const Default: Story = {
  args: {},
}
