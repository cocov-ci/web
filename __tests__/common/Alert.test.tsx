import { render, screen } from '@testing-library/react'
import { Ghost } from 'lucide-react'

import Alert from 'app/common/Alert'

import '@testing-library/jest-dom'

describe('common/Alert', () => {
  it('renders Alert with all params', () => {
    const { container } = render(
      <Alert
        description="description here.."
        icon={Ghost}
        title="title here.."
      />,
    )

    expect(screen.getByText('title here..')).toBeVisible()
    expect(screen.getByText('description here..')).toBeVisible()
    expect(container.querySelector('svg')).toBeVisible()
  })

  it('renders Alert only with "title" param', () => {
    const { container } = render(<Alert title="title here.." />)

    expect(container.querySelector('h2')).toBeTruthy()
    expect(container.querySelector('h3')).toBeFalsy()
    expect(container.querySelector('svg')).toBeFalsy()
  })

  it('renders Alert only with "description" param', () => {
    const { container } = render(<Alert description="description here.." />)

    expect(container.querySelector('h2')).toBeFalsy()
    expect(container.querySelector('h3')).toBeTruthy()
    expect(container.querySelector('svg')).toBeFalsy()
  })

  it('renders Alert only with "icon" param', () => {
    const { container } = render(<Alert icon={Ghost} />)

    expect(container.querySelector('h2')).toBeFalsy()
    expect(container.querySelector('h3')).toBeFalsy()
    expect(container.querySelector('svg')).toBeTruthy()
  })

  it('renders Alert snapshots', () => {
    const { container } = render(<Alert />)
    expect(container).toMatchSnapshot()
  })
})
