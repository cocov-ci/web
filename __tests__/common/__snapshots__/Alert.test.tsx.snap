// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert renders Alert snapshots 1`] = `
<div>
  <div
    class="alert"
  >
    <div
      class="box"
    >
      <div
        class="content"
      />
    </div>
  </div>
</div>
`;
